dotCMS/src/main/java/com/dotcms/graphql/business/RelationshipFieldGenerator.java

```diff
@@ -0,0 +1,122 @@
+                            + "." + field.variable(), e);
+        }
+
+        Relationship relationship;
+
+        try {
+            relationship = relationshipAPI.getRelationshipFromField(field,
+                    APILocator.systemUser());
+        } catch (DotDataException | DotSecurityException e) {
+            throw new DotRuntimeException(e);
+        }
+
+        final ContentletRelationships contentletRelationships = new ContentletRelationships(null);
+        final ContentletRelationships.ContentletRelationshipRecords
+                records = contentletRelationships.new ContentletRelationshipRecords(
+                relationship,
+                relationshipAPI.isChildField(relationship, field));
+
+        GraphQLOutputType outputType = GraphQLTypeReference.typeRef(relatedContentType.variable());
+
+        outputType = records.doesAllowOnlyOne()
+                ? outputType
+                : list(outputType);
+
+        return newFieldDefinition()
+                .name(field.variable())
+                .type(field.required() ? nonNull(outputType) : outputType)
+                .dataFetcher(new RelationshipFieldDataFetcher()).build();
+    }
+
+    private ContentType getRelatedContentTypeForField(final Field field) {
+        DotPreconditions.checkNotNull(field, "Field can't be null");
+        DotPreconditions.checkArgument(UtilMethods.isSet(field.variable()), "Field variable needs to be set");
+        DotPreconditions.checkArgument(UtilMethods.isSet(field.relationType()), "Field relationType needs to be set");
+
+        final Relationship relationship;
+        try {
+            relationship = relationshipAPI.getRelationshipFromField(field,
+                    APILocator.systemUser());
+        } catch (DotSecurityException | DotDataException e) {
+            throw new FieldGenerationException("Relationship with name:"
+                    + field.relationType() + " not found. Field var:" + field.variable()
+                    + ". Field ID: " + field.id());
+        }
+
+        if (relationship == null) {
+            throw new FieldGenerationException("Relationship with name:"
+                    + field.relationType() + " not found. Field var:" + field.variable()
+                    + ". Field ID: " + field.id());
+        }
+
+        final String relatedContentTypeId =
+                relationship.getParentStructureInode().equals(field.contentTypeId())
+                        ? relationship.getChildStructureInode()
+                        : relationship.getParentStructureInode();
+
+        ContentType type;
+
+        try {
+            type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(relatedContentTypeId);
+        } catch (DotSecurityException | DotDataException e) {
+            throw new FieldGenerationException("Unable to find content type with id:" + relatedContentTypeId);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)